### PR TITLE
Variables: Recognize only defined CLI options

### PR DESCRIPTION
**Pull request by medikoo**
_Not documented part of #8364_

With this update only recognized (by CLI command schema) options are resolved out of variables. It's an intended behavior of a new resolver from a beginning. It's a step towards direction, where we abort request if user passes unrecognized options (it's to avoid eventual errors, due to command not being used as intended)

Additionally:
- Resolve CLI command and options gradually (e.g. do not recognize AWS only commands, when we deal with no AWS provider)
- Fix resolution of `false` CLI param values (resolver returned `null`)
- Recognize `--stage` as any (not just AWS) service option. As it's crucial to load`.env`  files, a functionality that's provider agnostic
- Further improve commands schema, by recognizing `--version` on each command, and defining `help` command schema
- Avoid showing CLI options order deprecation with help request
- Expose resolved _command_ and _command schema_ on `resolveInput` result object
- Reorganize command CLI schema definitions, so they're grouped into three sections (_service independent_, _service dependent_, _AWS service dependent_)
- Validate `provider.name` at early stage.

### CODE REVIEW

File: lib/cli/commands-schema/aws-service.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
+'use strict';
+
+const awsServiceOptions = require('./common-options/service');
+
+const commands = (module.exports = new Map());
+
+commands.set('deploy', {
```
Comment (pgrzesik) [inline lib/cli/commands-schema/aws-service.js:7]: Should it be only available for `aws-service`? What about support for e.g. https://github.com/serverless/serverless-azure-functions ?

**Review by pgrzesik — COMMENTED** [review]
I have a general comment with `commands-schema/aws-service` - there's a few of commands defined such as `deploy`, `invoke` or `invoke local` that are not specific to AWS but are also supported e.g. by https://github.com/serverless/serverless-azure-functions. Is that breaking anything for these plugins?

**Comment by medikoo** [comment]
> I have a general comment with commands-schema/aws-service - there's a few of commands defined such as deploy, invoke or invoke local that are not specific to AWS but are also supported e.g. by https://github.com/serverless/serverless-azure-functions. Is that breaking anything for these plugins?

That's a good point, still let me explain reasoning for such separation.

Commands as configured in `aws-service` file, are commands which have zero logic configured in core if run in non-AWS provider.

They of course may work in other providers, but given provider is expected to configure some functionality for them, otherwise they will just do nothing.

e.g. in case of Azure, there's a logic configured for `sls deploy` -> https://github.com/serverless/serverless-azure-functions/blob/c53f535204c467339e8ae1f6edb985c3280bc32c/src/plugins/deploy/azureDeployPlugin.ts#L18

but e.g. Azure does not have any logic defined for `sls metrics` -> https://github.com/serverless/serverless-azure-functions/tree/master/src/plugins, it's also not existent in its docs. What currently happens is that Framework outputs `sls metrics` as valid command in help, but when you run it in context of Azure nothing happens.

This PR is first step to make that experience better. All commands configured in `aws-service` block, will have help displayed for in context of non-AWS provider, only if we find some hooks attached to those command lifecycle events, otherwise we will strip them from help.

Also I plan then to show help for those commands with AWS options stripped (as `app` or `org` etc)

Currently this PR doesn't change anything in how help is displayed (it's just internal organization of command schemas that was improved)

**Review by pgrzesik — APPROVED** [review]
Thanks for extensive clarification - looks great :+1:

**Pull request merged**